source/guides/non-denizen/resource-packs.md

```diff
@@ -0,0 +1,248 @@
+*Note: Color before formatting; Formatting codes persist after a color code, Not vise-versa!*
+
+You can find special characters in your Character Map. If you're on a Windows operating system, `Start` > `Windows Accessories`. If you're in Linux using GNOME and Unity, `Gucharmap character map` is a part of `GNOME desktop`. If you run a Gnome desktop - you can access it in any of these following ways:
+- Menu on the top of the screen, `(language)` > `Character Map`.
+- `Gucharmap` in terminal.
+- `Applications` > `Accessories` > `Character Map`.
+
+You can also google search for unicode characters.
+
+### Inside The Assets Folder Directory
+Directory: `.minecraft\resourcepacks\MyResourcePack\assets\`
+This directory should be empty except for the one folder directory: `minecraft`
+toss it in and leave everything else out of here.
+
+### Inside The Minecraft Folder Directory
+Directory: `.minecraft\resourcepacks\MyResourcePack\assets\minecraft\`
+
+Depending on what content you plan on changing, you need to opt creating these folders:
+
+- blockstates - This is where each block-state of items are saved.
+- font - This is where font data is saved. This guide does not cover this.
+- models - This is where the model data and files are saved
+- textures - This is where the texture image files are saved
+- fonts - This is where your font data is saved
+- sounds - This is where your sounds are saved
+- optifine - This is where your optifine data is saved. This guide does not cover this.
+
+For Optifine support, it's recommended you join their discord and review their documentation at their [Github Source](https://github.com/sp614x/optifine/tree/master/OptiFineDoc/doc).
+
+#### Example File: `sounds.json`
+This file indexes where Minecraft should look for your sounds. Below is an example of a setup for two custom sounds, `defence_levelup0` and `defence_levelup1`.
+```json
+{
+  "entity.player.defence.level": {
+    "sounds": [
+      {
+        "name": "custom/defence_levelup0"
+      },
+      {
+        "name": "custom/defence_levelup1"
+      }
+    ],
+    "subtitle": "Excited Trumpet Noises"
+  }
+}
+```
+
+The first key is the name for the command; in this example, `entity.player.defence.level`
+The only data object within the command we need to specify are `sounds`. Optionally, you can specify the subtitle that displays if subtitles are enabled in-game. If you place multiple sounds within the `"sounds":[]` array, the sound will randomize between them based on their weight.
+
+For each file, you will need the data: `"name":"FILEPATH/FILENAME"`, excluding the file's extension.
```

Comment:
Should specify what the file extension is/can be (`.ogg` always? Or are others valid?), and maybe add a part about converting file types if that's needed (`.ogg`, despite being the best audio file format, is not a very commonly seen one)